dbms/src/Functions/FunctionsHashing.h

```diff
@@ -82,6 +107,11 @@ struct HalfMD5Impl
 
         return Poco::ByteOrder::flipBytes(buf.uint64_data);        /// Compatibility with existing code.
     }
+
+    static UInt64 mergeHashes(UInt64 h1, UInt64 h2)
+    {
+        return IntHash64Impl::apply(h1) ^ h2;
```

Comment:
If the main hash function is cryptographic, we must use cryptographic hash function to combine too.

Reply:
How to do it, variants:
1. `combine(x, y) = sipHash64(x || y)` where `||` is "concatenation".
2. if there was more than one element in tuple, update (`SipHash::update`, `MD5_update`, etc.) state by length of each element then by contents of each element; and finalize at the end.